src/jsonToCsv.ts

```diff
@@ -0,0 +1,67 @@
+import { createReadStream, createWriteStream } from "fs";
+import { stringifyJsonChunk, prepareRow } from "./utils";
+import { Header, Paths, RowObject } from "./types";
+import JSONStream from "jsonstream";
+import { map } from "event-stream";
+import { Transform } from "stream";
+
+const encoding: BufferEncoding = "utf-8";
+
+const mapPipedData = map((data: RowObject, callback: Function) => {
+  const strigified = stringifyJsonChunk(data);
+  callback(null, strigified);
+});
+
+export const convertFilesWithJson = (header: Header, paths: Paths) => {
+  return new Promise((resolve, reject) => {
+    const pathSplitted = paths.in.split("/");
+    const fileName = pathSplitted[pathSplitted.length - 1];
+
+    try {
+      const reader = createReadStream(paths.in, encoding);
+      const writer = createWriteStream(paths.out, encoding);
+
+      writer.on("open", () => writer.write(prepareRow(header)));
+      writer.on("close", () =>
```

Comment:
There is no subscription to an error of your stream. Catch construction will not catch the error if something went wrong with stream.

Reply:
It's good to know that catch doesn't get the stream error. I added error event.